fairlearn/metrics/_annotated_metric_function.py

```diff
@@ -17,7 +20,7 @@
 _METRIC_FUNCTION_NOT_CALLABLE = "Object passed as metric function not callable"
 
 
-class AnnotatedMetricFunction:
+class AnnotatedMetricFunction(Generic[R]):
     """Wraps functions to make them callable with a DataFrame argument.
 
     The :class:`MetricFrame` makes extensive use of `pandas` DataFrames
```

Comment:
The class docstring might need an update, but I don't have the full picture yet, so I am seeking for support

Reply:
Update: Looking at the docs and at `fairlearn/metrics/__init__.py`, the only class to expose is `MetricFrame`. So I will try to update the docstrings here, knowing that it's not directly user facing